### PR TITLE
map_server: update to yaml-cpp 0.8.0

### PR DESCRIPTION
**Pull request by 981213**
yaml-cpp 0.8.0 requires explicit linking against yaml-cpp::yaml-cpp.
This fixes building breakage on rolling due to https://github.com/ros2/yaml_cpp_vendor/pull/48

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | None |
| Primary OS tested on | Ubuntu 24.04 |
| Robotic platform tested on | Not yet. It shouldn't affect anything pratically. |
| Does this PR contain AI generated software? | No |

---

## Description of contribution in a few bullet points

* Fix building on rolling due to recent yaml-cpp upgrade

## Description of documentation updates required from your changes

None.

---

## Future work that may be required in bullet points
None.

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by 981213** [comment]
The CI failed to build because it doesn't have the latest yaml_cpp_vendor package.

**Comment by SteveMacenski** [comment]
@981213 should this PR wait then until that lands?

File: nav2_map_server/CMakeLists.txt
Comment (SteveMacenski) [inline nav2_map_server/CMakeLists.txt:14]: This should not be necessary, this is the role of the vendor package.

File: nav2_map_server/CMakeLists.txt
Comment (SteveMacenski) [inline nav2_map_server/CMakeLists.txt:123]: Ditto and below

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Please follow up, I don't think this is an appropriate PR since we have the vendor. If the vendor needs to be updated, then update that and it'll be reflected here after a sync. Why are these changes needed here instead?

**Comment by 981213** [comment]
> should this PR wait then until that lands?

yaml_cpp_vendor 9.0.0 has been released into rolling, and that's how I noticed this problem. This is caused by the _vendor package update and independent of OS.

>  Why are these changes needed here instead - that breaks the vendorization?

This PR is created according to
https://github.com/ros2/yaml_cpp_vendor/pull/48#issuecomment-2046048236
and following:
https://github.com/ros-perception/image_common/pull/305
https://github.com/ros2/rviz/pull/1183
https://github.com/ros2/rosbag2/pull/1605

The solution provided upstream is to change all these downstream.

**Comment by 981213** [comment]
I agree that this change is a bit strange. Let me ask in that yaml-cpp update PR.

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Build has a number of complaints along the lines of 

```
CMake Error at CMakeLists.txt:32 (add_executable):
  Target "map_server" links to target "yaml-cpp::yaml-cpp" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

```

**Comment by 981213** [comment]
Oh, yaml_cpp_vendor 9.0 isn't available as binary package for jammy:
```
sudo apt-cache madison ros-rolling-yaml-cpp-vendor
ros-rolling-yaml-cpp-vendor | 8.3.1-1jammy.20240124.152723 | http://mirrors.ustc.edu.cn/ros2/ubuntu jammy/main amd64 Packages
```
```
sudo apt-cache madison ros-rolling-yaml-cpp-vendor
ros-rolling-yaml-cpp-vendor | 9.0.0-1noble.20240416.181051 | http://packages.ros.org/ros2/ubuntu noble/main amd64 Packages
```
This PR should be blocked until the switching to noble then.

**Comment by SteveMacenski** [comment]
OK! I'll tag it that way with the 24.04 PR, thanks for the help and proactivity!

**Comment by SteveMacenski** [comment]
Note to self that this is also a problem with Robot Localization to fix

**Comment by tonynajjar** [comment]
I ran into the same issue and sunk 2 hours to fix it in a similar way,if only I found this PR beforehand :sweat_smile: Anyway, I think this should now be ready to be merged

**Comment by SteveMacenski** [comment]
https://app.circleci.com/pipelines/github/ros-planning/navigation2/11466/workflows/31511139-b085-4f4b-ad71-5573452fa375/jobs/35332/steps

CI tells me this shouldn't happen until the 24.04 migration

**Comment by tonynajjar** [comment]
Ah I tried it with the rolling docker images and there it was needed to work. I guess CI is not using latest rolling yet

**Comment by SteveMacenski** [comment]
@981213 thanks for the contribution here - @tonynajjar is working on the full 24.04 migration which this commit was added to in https://github.com/ros-navigation/navigation2/pull/4298. So, I'm closing this as that PR supersedes with some other necessary changes. I want to thank you for your help here and responses! It is greatly appreciated - it all takes a village :-)

**Pull request closed without merging**
